### PR TITLE
#577 Visual feedback on coverage for the repo

### LINKED ISSUE #577: Visual feedback on coverage for the repo
Add codecov to the repo

### PR DESCRIPTION
**Pull request by biafra23**
Closes #577  .

Changes proposed in this pull request:
- Add jacoco code coverage for all modules
- Add badge to README.md

@gnosis/mobile-devs

### CODE REVIEW

File: contracts/build.gradle [hunk 1]
```diff
@@ -1,4 +1,5 @@
 apply from: '../buildsystem/android_library.gradle'
+apply from: '../buildsystem/coverageReport.gradle'
```
Comment (jpalvarezl) [inline contracts/build.gradle:2]: This module just contains generated files which won't be tested.

File: qrscanner/build.gradle [hunk 2]
```diff
@@ -2,6 +2,7 @@ apply from: '../buildsystem/android_library.gradle'
 apply from: '../buildsystem/constants.gradle'
 apply from: '../buildsystem/helpers.gradle'
 apply from: '../buildsystem/test_dependencies.gradle'
+apply from: '../buildsystem/coverageReport.gradle'
```
Comment (jpalvarezl) [inline qrscanner/build.gradle:5]: Like I mentioned in the standup, not sure if having the QR Scanning being covered with tests is a priority, considering the OKRs' definition. Would be nice if @rmeissner could provide some feedback here.

Comment (rmeissner) [reply qrscanner/build.gradle:5]: it is not considered for the OKR, but I would still generate the test coverage for it. In a "perfect" world we should have tests for it (e.g. to make sure that inverted QR codes work ;) )

Comment (rmeissner) [reply qrscanner/build.gradle:5]: Should we change the format of the `coverageReport.gradle` file name to snake case? (seems that you guys prefer this for file names ;) )

File: README.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
-# Gnosis Safe Android App [![Build Status](https://travis-ci.org/gnosis/safe-android.svg?branch=master)](https://travis-ci.org/gnosis/safe-android)
+# Gnosis Safe Android App 
+[![Build Status](https://travis-ci.org/gnosis/safe-android.svg?branch=master)](https://travis-ci.org/gnosis/safe-android)
+[![codecov](https://codecov.io/gh/gnosis/safe-android/branch/master/graph/badge.svg)](https://codecov.io/gh/gnosis/safe-android)
```
Comment (rmeissner) [inline README.md:3]: 14% 😢

Comment (rmeissner) [reply README.md:3]: It seems not all tests are considered. E.g. codecov says that the settings view model is not tests, but there are tests for it https://github.com/gnosis/safe-android/blob/master/app/src/test/java/io/gnosis/safe/ui/safe/settings/SafeSettingsViewModelTest.kt

Could we investigate this.

File: qrscanner/build.gradle (hunk 2 above)
Comment (jpalvarezl) [reply qrscanner/build.gradle:5]: I agree with changing the casing too

File: contracts/build.gradle (hunk 1 above)
Comment (biafra23) [reply contracts/build.gradle]: Fixed in 0a133dbfa206012fcdaa9a90c4a5c8b51a9d7331

File: qrscanner/build.gradle (hunk 2 above)
Comment (biafra23) [reply qrscanner/build.gradle]: Done in 077d4e65041543446b06c1307ecf89d3dc4d2807

File: README.md (hunk 3 above)
Comment (biafra23) [reply README.md:3]: According to this it has some coverage: https://codecov.io/gh/gnosis/safe-android/pull/604/src/app/src/main/java/io/gnosis/safe/ui/safe/settings/SafeSettingsViewModel.kt

Comment (biafra23) [reply README.md:3]: That is the same coverage that I see in `app/build/reports/jacoco/createDebugTestCoverage/html/io.gnosis.safe.ui.safe.settings/SafeSettingsViewModel.kt.html` locally. After running: `./gradlew createDebugTestCoverage`

Comment (rmeissner) [reply README.md:3]: let's see if that is fixed for master after the merge :)

File: README.md [hunk 4]
```diff
@@ -1,4 +1,6 @@
-# Gnosis Safe Android App [![Build Status](https://travis-ci.org/gnosis/safe-android.svg?branch=master)](https://travis-ci.org/gnosis/safe-android)
+# Gnosis Safe Android App 
+[![Build Status](https://travis-ci.org/gnosis/safe-android.svg?branch=master)](https://travis-ci.org/gnosis/safe-android)
```
Comment (elgatovital) [inline README.md:2]: we don't need travis

Comment (jpalvarezl) [reply README.md:2]: What do we need this line for?

Comment (biafra23) [reply README.md]: Removed in e3d6fa7ee4d901fb59cc67e34249d9bd4300ab8d

**Pull request merged**